app/controllers/admin/ministries_controller.rb

```diff
@@ -36,14 +59,35 @@ def new
 	end
 
 	def create
-    @ministry = Ministry.new(secure_params.merge(created_by_id: current_user.id))
-    if @ministry.save
-      redirect_to admin_ministry_path(@ministry), flash_success_info: "Ministry was successfully created."
-    else
-    	flash[:flash_info] = "Ministry was not successfully created."
-      render :new
-    end
-  end
+		@ministry = Ministry.new(secure_params.merge(created_by_id: current_user.id))
+		
+		if @ministry.location_id.nil?
+			@ministry.location_id = 0
```

Comment:
why is `0` being used?

Reply:
The 'Location' model currently supports only states and cities, towns  or villages. This is fine for tagging locations to state level ministries and local level ministries. But national level ministries cannot be tagged to these locations. Hence we have decided to use a value of 0 without breaking any of the existing functionality with respect to location. National level ministries cannot be assigned a location and will show the value as 'None' in the views. The validations also ensure that a location cannot be selected for a national level ministry.